invenio/modules/search/walkers/elasticsearch.py

```diff
@@ -113,4 +157,43 @@ def visit(self, node):
             "match_all": {}
         }
 
+    @visitor(SpiresOp)
+    def visit(self, node, keyword, value_fn):
+        kw = self.handle_spires_kw(keyword)
+        if callable(value_fn):
+            return value_fn(kw)
+        raise RuntimeError("Not supported second level operation.")
+
+    @staticmethod
+    def _operators(node, value_fn):
+        operands = {GreaterOp: "gt", LowerOp: "lt", GreaterEqualOp: "gte",
+                    LowerEqualOp: "lte"}
+        value = value_fn(None)["multi_match"]["query"]
+        condition = {operands[type(node)]: value}
+
+        def _f(keyword):
+            if len(keyword) > 1:
+                res = {"bool": {"should": []}}
+                res["bool"]["should"] = [{'range': {k: condition}}
+                                         for k in keyword]
+            else:
+                res = {'range': {keyword[0]: condition}}
+            return res
+        return _f
+
+    @visitor(GreaterOp)
+    def visit(self, node, value_fn):
+        return self._operators(node, value_fn)
```

Comment:
Can you move some logic from `_operands` to visitors? The visitors are doing to mapping that it's done here https://github.com/inveniosoftware/invenio/pull/2956/files#diff-d96416206488cc09b0c53ea3907f8d84R169

Example:
```python
@staticmethod
def operands(node, operand, value):
    ...

@visitor(LowerOp)
def visit(self, node, value):
    return self._operators(node, 'lt', value()["multi_match"]["query"])
```